python/thunder/extraction/source.py

```diff
@@ -106,18 +109,66 @@ def restore(self, skip=None):
                 del self.__dict__[prop]
         return self
 
-    def distance(self, other):
+    def distance(self, other, method='l1'):
         """
         Distance between the center of this source and another.
+
+        Parameters
+        ----------
+        method : str
+            Specify a distance measure to used for spatial distance between source
+            centers. Current options include Euclidean distance ('euclidean') and 
+            L1-norm ('l1'). 
+
         """
+        if method == 'euclidean':
+            order = 2
+        elif method == 'l1':
+            order = 1
+        else:
+            raise ValueError('Given distance method not supported.')
+
         if isinstance(other, Source):
-            return sqrt(sum((self.center - other.center) ** 2))
+            return la.norm(self.center - other.center, ord=order)
         elif isinstance(other, list) or isinstance(other, ndarray):
-            return sqrt(sum((self.center - asarray(other)) ** 2))
+            return la.norm(self.center - asarray(other), ord=order)
+
+    def overlap(self, other, method='support',counts=False):
+        """
+        Compute the overlap between current source and 'other' source, in terms 
+        of either support or similarity of coefficients. 
+
+        Parameters
+        ----------
+        method : str
+            Compare either support of source coefficients ('support'), or the 
+            source spatial filters (not yet implemented).
+
+        """
+        if isinstance(self.coordinates, ndarray):
+            self.coordinates = self.coordinates.tolist()
+        if isinstance(other.coordinates, ndarray):
+            other.coordinates = other.coordinates.tolist()
+        intersection = [a for a in self.coordinates if a in other.coordinates]
+        complement = [a for a in self.coordinates if a not in intersection]
+        if method == 'support':
+            hits = len(intersection)
+            misses = len(complement)
+            if counts:
+                return (hits,misses)
+            else:
+                return hits/float(hits+misses)
+        elif method == 'corr':
```

Comment:
Or intersection? Specifically, take the intersection computed above, and compute a correlation coefficient on just those values?